Client/Frontend/Home/TopSitesPanel.swift

```diff
@@ -136,16 +138,18 @@ class TopSitesPanel: UIViewController {
     }
 
     private func deleteHistoryTileForSite(site: Site, atIndexPath indexPath: NSIndexPath) {
-        func reloadThumbnails() {
-            self.profile.history.getTopSitesWithLimit(self.layout.thumbnailCount)
-                .uponQueue(dispatch_get_main_queue()) { result in
+        func reloadThumbnails() -> Success {
+            return self.profile.history.getTopSitesWithLimit(self.layout.thumbnailCount)
+                .bindQueue(dispatch_get_main_queue()) { result in
                     self.deleteOrUpdateSites(result, indexPath: indexPath)
```

Comment:
Would it be possible to do the deletion and replacement separately? That is:
 1. Immediately delete the item from the collection.
 2. Make queries to delete the item and get the updated data.
 3. Fill in top sites with new thumbnails.

Right now, I know that 1 and 3 happen in the same operation as part of `performBatchUpdates`, which I think we do because of animations. Is there any way we can split them, doing the deletion before any querying, or does delete/replace need to be atomic?